cdc/sinkv2/eventsink/txn/worker.go

```diff
@@ -160,47 +165,51 @@ func (w *worker) runBackgroundLoop() {
 
 func (w *worker) onEvent(txn txnWithNotifier) bool {
 	w.metricConflictDetectDuration.Observe(time.Since(txn.start).Seconds())
+	w.metricTxnWorkerHandledRows.Add(float64(len(txn.Event.Rows)))
 	w.wantMoreCallbacks = append(w.wantMoreCallbacks, txn.wantMore)
 	if w.backend.OnTxnEvent(txn.txnEvent.TxnCallbackableEvent) {
-		if !w.timer.Stop() {
-			<-w.timer.C
-		}
 		return true
 	}
 	return false
 }
 
 // doFlush flushes the backend. Returns true if the goroutine can exit.
-func (w *worker) doFlush(flushTimeSlice *time.Duration) bool {
-	start := time.Now()
-	defer func() {
-		elapsed := time.Since(start)
-		*flushTimeSlice += elapsed
-		w.metricTxnWorkerFlushDuration.Observe(elapsed.Seconds())
-	}()
+func (w *worker) doFlush(flushTimeSlice *time.Duration, needStopTimer bool) bool {
```

Comment:
Seems that the signature of this method is not easy to understand. Can you add some comments?